src/buildHTML.js

```diff
@@ -74,24 +91,18 @@ const buildExpression = function(expression, options, isRealGroup) {
     // spacing (e.g., "add thick space between mord and mrel"). Since CSS
     // adjacency rules implement atom spacing, spaces should be invisible to
     // CSS. So we splice them out of `groups` and into the atoms themselves.
-    let spaces = null;
     for (let i = 0; i < groups.length; i++) {
-        if (isSpace(groups[i])) {
-            spaces = spaces || [];
-            spaces.push(groups[i]);
-            groups.splice(i, 1);
-            i--;
-        } else if (spaces) {
+        const spaces = spliceSpaces(groups, i);
+        if (spaces && i < groups.length) {
```

Comment:
I'm sorry that you felt your time was disrespected -- not my intent.  We certainly appreciate the time you've spent in fixing this bug (and in implementing the original feature).  I'll move this agreed-to-be stylistic issue to a separate PR.